tensorflow/contrib/android/java/org/tensorflow/contrib/android/TensorFlowInferenceInterface.java

```diff
@@ -158,6 +158,8 @@ public TensorFlowInferenceInterface(InputStream is) {
       }
     } catch (IOException e) {
       throw new RuntimeException("Failed to load model from the input stream", e);
+    } finally {
+       is.close(); 
```

Comment:
Oh, wait, my apologies - I had misread this change.
It seems you're changing the contract of the public `TensorFlowInferenceInterface(InputStream)` constructor so that it "takes ownership" of the `InputStream`.

Could you elaborate on why we'd want to change the contract? It seems more reasonable to me that the creator of the `InputStream` be responsible for closing it. Am I missing something?

Reply:
`It seems you're changing the contract of the public ` I am sorry I didn't get you. Regarding reasonable, `InputStream ` owns resources that must be explicitly freed by invoking `is.close()` and yes the creator of the `InputStream` be responsible for closing it.

Reply:
Let me rephrase.

The constructor is `public TensorFlowInferenceInterface(InputStream is)`.
Without this PR, the constructor would not call `is.close()`. Which means that client code would look like this:

```java
InputStream s = createInputStream();
TensorFlowInferenceInterface tf = new TensorFlowInferenceInterface(s);
// Perhaps do something else with s
s.close()
```

In this PR, you're changing the constructor to call `is.close()`, which means that the creation and closing of the `InputStream` is asymmetric. For example:

```java
InputStream s = createInputStream();
TensorFlowInferenceInterface tf = new TensorFlowInferenceInterface(s);
// s can no longer be used by the client as it is  closed.
```

I don't think we should be making that change.

Reply:
Ok got it!, just thought in case. In future, if we you are using that method. So it could be beneficial if we are closing it with final clause within the method itself. However, I don't think that is going to throw an error.

Reply:
I didn't quite follow what you meant by "it could be beneficial if we are closing it with final clause within the method itself", but I gather we're on the same page, so closing this PR.

Thanks!